src/watchdog/observers/inotify_c.py

```diff
@@ -317,6 +317,15 @@ def _recursive_simulate(src_path):
                         del self._wd_for_path[move_src_path]
                         self._wd_for_path[inotify_event.src_path] = moved_wd
                         self._path_for_wd[moved_wd] = inotify_event.src_path
+                        if self.is_recursive and inotify_event.is_directory:
+                            for path in self._path_for_wd.itervalues():
+                                if path.startswith(move_src_path):
```

Comment:
You need make sure you append `/` when doing subdirectory checks. Otherwise it will catch directories that starts with the same name.